eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -5,84 +5,148 @@
+    [string]$Alias
+  )
 
-    if ($OwnerAliasCache.ContainsKey($Alias)) {
-      return $OwnerAliasCache[$Alias]
-    }
+  if ($OwnerAliasCache.ContainsKey($Alias)) {
+    return $OwnerAliasCache[$Alias]
+  }
 
-    $domains = @("microsoft.com", "ntdev.microsoft.com")
+  # AAD apps require a higher level of permission requiring admin consent to query the MS Graph list API
+  # https://docs.microsoft.com/en-us/graph/api/user-list?view=graph-rest-1.0&tabs=http#permissions
+  # The Get-AzAdUser call uses the list API under the hood (`/users/$filter=<alias>`)
+  # and for some reason the Get API (`/user/<id or user principal name>`) also returns 401
+  # with User.Read and User.ReadBasic.All permissions when called with an AAD app.
+  # For this reason, skip trying to query MS Graph directly in provisioner mode.
+  # The owner alias cache should already be pre-populated with all user records from the
+  # github -> ms alias mapping retrieved via the repos.opensource.microsoft.com API, however
+  # this will not include any security groups, in the case an owner tag does not contain
+  # individual user aliases.
+  if ($IsProvisionerApp) {
+    Write-Host "Skipping MS Graph alias lookup for '$Alias' due to permissions. Owner aliases not registered with github will be treated as invalid."
+    $OwnerAliasCache[$Alias] = $false
+    return $false
+  }
 
-    foreach ($domain in $domains) {
-        if (Get-AzAdUser -UserPrincipalName "$Alias@$domain") {
-            $OwnerAliasCache[$Alias] = $true
-            return $true;
-        }
+  $domains = @("microsoft.com", "ntdev.microsoft.com")
+
+  foreach ($domain in $domains) {
+    if (Get-AzAdUser -UserPrincipalName "$Alias@$domain") {
+      $OwnerAliasCache[$Alias] = $true
+      return $true;
     }
+  }
 
-    $OwnerAliasCache[$Alias] = $false
+  $OwnerAliasCache[$Alias] = $false
 
-    return $false;
+  return $false;
+}
+
+function AddGithubUsersToAliasCache() {
+  if ($GithubAliasCachePath -and (Test-Path $GithubAliasCachePath)) {
+    Write-Host "Loading github -> microsoft alias mappings from filesystem cache."
+    $users = Get-Content $GithubAliasCachePath | ConvertFrom-Json -AsHashtable
+  } else {
+    Write-Host "Retrieving github -> microsoft alias mappings from opensource API."
+    $users = GetAllGithubUsers $OpensourceApiApplicationTenantId $OpensourceApiApplicationId $OpensourceApiApplicationSecret
+  }
+  if (!$users) {
+    Write-Error "Failed to retrieve github -> microsoft alias mappings from opensource api."
+    exit 1
+  }
+  foreach ($user in $users) {
+    if ($user.aad.alias) {
+      $OwnerAliasCache[$user.aad.alias] = $true
+    }
+  }
+  Write-Host "Found $($OwnerAliasCache.Count) github+microsoft user records."
+  if ($GithubAliasCachePath -and !(Test-Path $GithubAliasCachePath)) {
+      Write-Host "Caching github -> microsoft alias mappings to '$GithubAliasCachePath'"
+      $users | ConvertTo-Json -Depth 100 | Out-File $GithubAliasCachePath -WhatIf:$false
```

Comment:
Nit: Could we actually limit depth or even trim out what we don't need just to load faster next time?

Reply:
I think depth is actually just 3 or 4. I can change it to what we need.